src/main/java/com/microsoft/sqlserver/jdbc/SQLServerPreparedStatement.java

```diff
@@ -2554,6 +2620,368 @@ public final void addBatch() throws SQLServerException {
+                tempint = localUserSQL.indexOf("\"", tempint + 1);
+            }
+            
+            // we've found a " that is actually trying to close the quote.
+            String tempstr = localUserSQL.substring(1, tempint);
+            localUserSQL = localUserSQL.substring(tempint + 1);
+            listOfColumns.add(tempstr);
+            return parseUserSQLForColumnListDWHelper(listOfColumns);
+        }
+
+        // At this point, the next chunk of string is the column name, without starting with [ or ".
+        StringBuilder sb = new StringBuilder();
+        while (localUserSQL.length() > 0) {
+            if (localUserSQL.charAt(0) == ',' || localUserSQL.charAt(0) == ')') {
+                if (localUserSQL.charAt(0) == ',') {
+                    localUserSQL = localUserSQL.substring(1);
+                    listOfColumns.add(sb.toString());
+                    return parseUserSQLForColumnListDWHelper(listOfColumns);
+                } else {
+                    localUserSQL = localUserSQL.substring(1);
+                    listOfColumns.add(sb.toString());
+                    return listOfColumns;
+                }
+            } else if (localUserSQL.substring(0, 2).equalsIgnoreCase("/*")) {
+                int temp = localUserSQL.indexOf("*/") + 2;
+                localUserSQL = localUserSQL.substring(temp);
+                localUserSQL = localUserSQL.trim();
+            } else {
+                sb.append(localUserSQL.charAt(0));
+                localUserSQL = localUserSQL.substring(1);
+            }
+        }
+        
+        // It shouldn't come here. If we did, something is wrong.
+        throw new IllegalArgumentException("localUserSQL");
+    }
+    
+
+    private ArrayList<String> parseUserSQLForValueListDW(boolean hasValuesBeenFound) {
+        localUserSQL = localUserSQL.trim();
+        
+        // ignore all comments
+        if (localUserSQL.substring(0, 2).equalsIgnoreCase("/*")) {
+            int temp = localUserSQL.indexOf("*/") + 2;
+            localUserSQL = localUserSQL.substring(temp);
+            return parseUserSQLForValueListDW(hasValuesBeenFound);
+        }
+        
+        if (localUserSQL.substring(0, 2).equalsIgnoreCase("--")) {
+            int temp = localUserSQL.indexOf("\n") + 2;
+            localUserSQL = localUserSQL.substring(temp);
+            return parseUserSQLForValueListDW(hasValuesBeenFound);
+        }
+        
+        if (!hasValuesBeenFound) {
+            // look for keyword "VALUES"
+            if (localUserSQL.substring(0, 6).equalsIgnoreCase("VALUES")) {
+                localUserSQL = localUserSQL.substring(6);
+                
+                localUserSQL = localUserSQL.trim();
+                
+                // ignore all comments
+                if (localUserSQL.substring(0, 2).equalsIgnoreCase("/*")) {
+                    int temp = localUserSQL.indexOf("*/") + 2;
```

Comment:
This logic is done quite a lot. Can we maybe make a private function that takes in a string and returns what we need. So the final result for all these if blocks would be something like
`if(localUserSQL.substring(0, 2).equalsIgnoreCase("/*")) {
    return newPrivateFunction("*/");
}`